src/ray/gcs/client.cc

```diff
@@ -2,44 +2,135 @@
 
 #include "ray/gcs/redis_context.h"
 
+static void GetRedisShards(redisContext *context, std::vector<std::string> *addresses,
+                           std::vector<int> *ports) {
```

Comment:
Can we change these pointers to references?

Reply:
Done